### PR TITLE
docs: added identity-center.md doc for AWS SSO

### PR DESCRIPTION
**Pull request by zeusal**
I have added specific documentation for configuring Argo CD with Identity Center (AWS SSO). #14703 

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.

This PR adds documentation to the User Management section to document integration of Identity Center SSO with ArgoCD as proposed in this discussion: https://github.com/argoproj/argo-cd/issues/14703

### CODE REVIEW

File: docs/operator-manual/user-management/identity-center.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  url: https://argocd.example.com
+```
+1. Configure Attribute mappings
+    * ![Identity Center SAML App 4](../../assets/identity-center-4.png)
+    * ![Identity Center SAML App 5](../../assets/identity-center-5.png)
+
+<!-- markdownlint-disable MD046 -->
+
+1. Download the CA certificate to use in the `argocd-cm` configuration.
+    * If you are using this in the caData field, you will need to pass the entire certificate (including `-----BEGIN CERTIFICATE-----` and `-----END CERTIFICATE-----` stanzas) through base64 encoding, for example, `base64 my_cert.pem`.
```
Comment (csantanapr) [inline docs/operator-manual/user-management/identity-center.md:31]:
```suggestion
    * If using the `caData` field, you'll need to base64-encode the entire certificate, including the `-----BEGIN CERTIFICATE-----` and `-----END CERTIFICATE-----` stanzas (e.g., `base64 my_cert.pem`).
```

File: docs/operator-manual/user-management/identity-center.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    * If you are using the ca field and storing the CA certificate separately as a secret, you will need to mount the secret to the `dex` container in the `argocd-dex-server` Deployment.
```
Comment (csantanapr) [inline docs/operator-manual/user-management/identity-center.md:32]:
```suggestion
    * If using the `ca` field and storing the CA certificate separately as a secret, you will need to mount the secret onto the `dex` container in the `argocd-dex-server` Deployment.
```

File: docs/operator-manual/user-management/identity-center.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      entityIssuer: https://external.path.to.argocd.io/api/dex/callback
+      redirectURI: https://external.path.to.argocd.io/api/dex/callback
+      usernameAttr: email
+      emailAttr: email
+      groupsAttr: groups
+```
+<!-- markdownlint-enable MD046 -->
+
+### Connect Identity Center Groups to Argo CD Roles
+Argo CD is aware of user memberships of Identity Center groups that match the *Group Attribute Statements* regex.
```
Comment (csantanapr) [inline docs/operator-manual/user-management/identity-center.md:61]:
```suggestion
Argo CD recognizes user memberships in Identity Center groups that match the **Group Attribute Statements** regex. 

```

File: docs/operator-manual/user-management/identity-center.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+The example above uses the `argocd-*` regex, so Argo CD would be aware of a group named `argocd-admins`.
```
Comment (csantanapr) [inline docs/operator-manual/user-management/identity-center.md:62]:
```suggestion
 In the example above, the regex `argocd-*` is used, making Argo CD aware of a group named `argocd-admins`.
```

**Review by csantanapr — CHANGES_REQUESTED** [review]
thank you for the PR @zeusal, some minor nits to fix

**Comment by ankursrc** [comment]
Hi @csantanapr @zeusal,
I would like to highlight that official AWS Identity Center documentation does not mention `user:groups` attribute . This is not supported as of now. Only below attributes are supported for application integration. 

${user:AD_GUID}
${user:email}
${user:familyName}
${user:givenName}
${user:middleName}
${user:name}
${user:preferredUsername}
${user:subject}

Link : https://docs.aws.amazon.com/singlesignon/latest/userguide/attributemappingsconcept.html#supportedssoattributes

**Comment by zeusal** [comment]
> thank you for the PR @zeusal, some minor nits to fix

@csantanapr Thanks for the comments, I will apply them to my commit. 

> Hi @csantanapr @zeusal, I would like to highlight that official AWS Identity Center documentation does not mention `user:groups` attribute . This is not supported as of now. Only below attributes are supported for application integration.
> 
> ${user:AD_GUID} ${user:email} ${user:familyName} ${user:givenName} ${user:middleName} ${user:name} ${user:preferredUsername} ${user:subject}
> 
> Link : https://docs.aws.amazon.com/singlesignon/latest/userguide/attributemappingsconcept.html#supportedssoattributes

@ankursrc  I know that the official documentation does not indicate that this attribute exists, but it does. If you have the possibility try it, I am currently using it for both Argocd and Jenkins. 
https://repost.aws/questions/QUKn8D8aQUSTuxyjcEwfICow/is-it-possible-to-map-the-group-as-an-attribute

**Review by morey-tech — APPROVED** [review]

**Comment by 34fathombelow** [comment]
> > thank you for the PR @zeusal, some minor nits to fix
> 
> @csantanapr Thanks for the comments, I will apply them to my commit.
> 
> > Hi @csantanapr @zeusal, I would like to highlight that official AWS Identity Center documentation does not mention `user:groups` attribute . This is not supported as of now. Only below attributes are supported for application integration.
> > ${user:AD_GUID} ${user:email} ${user:familyName} ${user:givenName} ${user:middleName} ${user:name} ${user:preferredUsername} ${user:subject}
> > Link : https://docs.aws.amazon.com/singlesignon/latest/userguide/attributemappingsconcept.html#supportedssoattributes
> 
> @ankursrc I know that the official documentation does not indicate that this attribute exists, but it does. If you have the possibility try it, I am currently using it for both Argocd and Jenkins. https://repost.aws/questions/QUKn8D8aQUSTuxyjcEwfICow/is-it-possible-to-map-the-group-as-an-attribute
> 
> The proof that it works: image image

@zeusal Can we mention that this is not officially supported in the AWS docs, however the workaround is currently working?

**Comment by zeusal** [comment]
> > > thank you for the PR @zeusal, some minor nits to fix
> > 
> > 
> > @csantanapr Thanks for the comments, I will apply them to my commit.
> > > Hi @csantanapr @zeusal, I would like to highlight that official AWS Identity Center documentation does not mention `user:groups` attribute . This is not supported as of now. Only below attributes are supported for application integration.
> > > ${user:AD_GUID} ${user:email} ${user:familyName} ${user:givenName} ${user:middleName} ${user:name} ${user:preferredUsername} ${user:subject}
> > > Link : https://docs.aws.amazon.com/singlesignon/latest/userguide/attributemappingsconcept.html#supportedssoattributes
> > 
> > 
> > @ankursrc I know that the official documentation does not indicate that this attribute exists, but it does. If you have the possibility try it, I am currently using it for both Argocd and Jenkins. https://repost.aws/questions/QUKn8D8aQUSTuxyjcEwfICow/is-it-possible-to-map-the-group-as-an-attribute
> > The proof that it works: image image
> 
> @zeusal Can we mention that this is not officially supported in the AWS docs, however the workaround is currently working?

@34fathombelow that sounds good, I can write it as a note.

**Comment by zeusal** [comment]
@34fathombelow Suggestion added as a note.

File: docs/operator-manual/user-management/identity-center.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+1. Copy the Argo CD URL into the `data.url` field in the `argocd-cm` ConfigMap.
+
+<!-- markdownlint-disable MD046 -->
+```yaml
+data:
+  url: https://argocd.example.com
+```
+1. Configure Attribute mappings.
+   > [!NOTE]
+   > Group attribute mapping is not officially supported in the AWS docs, however the workaround is currently working.
```
Comment (34fathombelow) [inline docs/operator-manual/user-management/identity-center.md:26]:
```suggestion
    !!! note
    Group attribute mapping is not officially supported in the AWS docs, however the workaround is currently working.
```

**Review by 34fathombelow — CHANGES_REQUESTED** [review]
One last Nit

**Review by gdsoumya — APPROVED** [review]
LGTM

File: docs/operator-manual/user-management/identity-center.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+<!-- markdownlint-disable MD046 -->
+```yaml
+data:
+  url: https://argocd.example.com
+```
+1. Configure Attribute mappings.
+    !!! note
+    Group attribute mapping is not officially supported in the AWS docs, however the workaround is currently working.
+    * ![Identity Center SAML App 4](../../assets/identity-center-4.png)
+    * ![Identity Center SAML App 5](../../assets/identity-center-5.png)
```
Comment (34fathombelow) [inline docs/operator-manual/user-management/identity-center.md:28]: @zeusal My apologies this is still not rendering the docs correctly.
Screenshot 2023-10-26 at 10 23 55 AM

I tested this locally and it works as expected:
```yaml
1. Configure Attribute mappings.

!!! note
    Group attribute mapping is not officially supported in the AWS docs, however the workaround is currently working.

  * ![Identity Center SAML App 4](../../assets/identity-center-4.png)
  * ![Identity Center SAML App 5](../../assets/identity-center-5.png)
  ```

**Review by 34fathombelow — CHANGES_REQUESTED** [review]

File: docs/operator-manual/user-management/identity-center.md (hunk 6 above)
Comment (zeusal) [reply docs/operator-manual/user-management/identity-center.md]: @34fathombelow I've changed it again, I think it's fine.

File: docs/operator-manual/user-management/identity-center.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+<!-- markdownlint-disable MD046 -->
+```yaml
+data:
+  url: https://argocd.example.com
+```
+
+4. Configure Attribute mappings.
+
+    !!! note
+       Group attribute mapping is not officially supported in the AWS docs, however the workaround is currently working.
```
Comment (34fathombelow) [inline docs/operator-manual/user-management/identity-center.md:28]: Sorry about how the docs are rendered.   Please see lines #2-5 and note the spacing and indentations.
```yaml
4. Configure Attribute mappings.

!!! note
    Group attribute mapping is not officially supported in the AWS docs, however the workaround is currently working.

  * ![Identity Center SAML App 4](../../assets/identity-center-4.png)
  * ![Identity Center SAML App 5](../../assets/identity-center-5.png)
 ```

**Review by 34fathombelow — CHANGES_REQUESTED** [review]

File: docs/operator-manual/user-management/identity-center.md (hunk 7 above)
Comment (zeusal) [reply docs/operator-manual/user-management/identity-center.md:28]: I am trying to make the change that you suggested but it doesn't look good. I am attaching an image to see the differences.

In red is the change you suggested and in yellow is the actual code. The rest of the document looks the same.
IMG_20231026_211947

Comment (34fathombelow) [reply docs/operator-manual/user-management/identity-center.md:28]: You can run locally `make build-docs` & `make serve-docs` then visit localhost:3000 to see the changes you have made.   It is also possible to see the changes you made to the docs after a commit by clicking the Details section of `docs/readthedocs.org:argo-cd` in the CI checks.
Screenshot 2023-10-26 at 1 22 17 PM

After making the changes to the `!!! note` section, it should be rendered similar to this:

Screenshot 2023-10-26 at 1 29 24 PM


Thank you so much for your patience.  If its to frustrating I will go ahead an accept the PR as is, then submit a fix for it.  Let me know if you would like to continue to fix this.

Comment (zeusal) [reply docs/operator-manual/user-management/identity-center.md]: @34fathombelow  Thank you very much for your help and your patience.

I have already understood you, I think that the format of the note is ok, however, I can't get the order of the numerical list right.

Comment (34fathombelow) [reply docs/operator-manual/user-management/identity-center.md]: No problem!  It looks like this code block is causing it:

```yaml
data:
  url: https://argocd.example.com
```
Most of our documentation uses headers vs numbered list.  Nothing wrong with the standard way you did it, but  I was able to come up with a workaround by indenting the code block by 4 tabs.   You have done enough work, we can fix this later.  If you want to make the changes feel free, if not I will merge as is later today.

Screenshot 2023-10-27 at 4 31 03 AM

Comment (zeusal) [reply docs/operator-manual/user-management/identity-center.md]: I have applied your suggestion, although it still doesn't follow the order, maybe the best thing to do is to merge it like this and then solve the visualization.

**Review by 34fathombelow — APPROVED** [review]
LGTM, thanks for all your hard work and patience.

**Comment by csantanapr** [comment]
@34fathombelow Thank you so much for reviewing this PR 🙏 

@zeusal get in touch with me on slack to work on getting the AWS docs updated in the mean time we can can have the note, when it's in the AWS docs we can come back remove it from argocd docs

**Comment by csantanapr** [comment]
Thank you @morey-tech for reviewing this PR also 🙏

**Review by blakepettersson — APPROVED** [review]

**Review by crenshaw-dev — APPROVED** [review]

**Comment by geek0ps** [comment]
@zeusal Please how can i set this up? I have been trying to set it up but without any progress?

**Comment by zeusal** [comment]
> @zeusal Please how can i set this up? I have been trying to set it up but without any progress?

Can you write your actual configuration and logs here? Please.

**Comment by geek0ps** [comment]
> > @zeusal Please how can i set this up? I have been trying to set it up but without any progress?
> 
> Can you write your actual configuration and logs here? Please.

can I write you an email or maybe hop on a call with you?

**Comment by zeusal** [comment]
> below is my configuration
> 
> ```
> configs:
>   params:
>     "server.insecure": true
>   cm:
>     create: true
>     url: "https://${hostname}"
>     exec.enabled: false
> %{ if sso_enabled ~}
>     dex.config: |
>       logger:
>         level: debug
>         format: json
>       connectors:
>         - type: saml
>           id: aws
>           name: "AWS SSO"
>           config:
>             ssoURL: ${aws_sso_url}
>             caData: ${aws_sso_cert}
>             entityIssuer: https://${hostname}/api/dex/callback
>             redirectURI: https://${hostname}/api/dex/callback
>             usernameAttr: subject
>             emailAttr: email
>             groupsAttr: groups
> %{ endif ~}
>   rbac:
>     create: true
>     policy.default: 'role:readonly'
>     policy.csv: |
>       p, role:org-admin, applications, *, */*, allow
>       p, role:org-admin, clusters, get, *, allow
>       p, role:org-admin, repositories, get, *, allow
>       p, role:org-admin, repositories, create, *, allow
>       p, role:org-admin, repositories, update, *, allow
>       p, role:org-admin, repositories, delete, *, allow
>       p, role:org-admin, projects, get, *, allow
>       p, role:org-admin, projects, create, *, allow
>       p, role:org-admin, projects, update, *, allow
>       p, role:org-admin, projects, delete, *, allow
>       p, role:org-admin, logs, get, *, allow
>       p, role:org-admin, exec, create, */*, allow
>       g, ${aws_sso_group_id}, role:org-admin
>     scopes: "[groups,email]"
>     ```
>     
>     after doing this i get the following error:  bad input
>     
>     @zeusal 
> ```

Hello @geek0ps  sorry for delay to answer.

I think it's because you're adding ${aws_sso_group_id} without quotes, Can you try adding double quotes to this value?

Regards!

**Pull request merged**